templates/examples.go

```diff
@@ -15,48 +16,73 @@ import (
 )
 
 var (
-	//go:embed templates
+	//go:embed quickstart
 	files embed.FS
 
-	exampleBasePath = "https://github.com/coder/coder/tree/main/examples/templates/"
+	exampleBasePath = "https://github.com/coder/coder/tree/main/examples"
 	examples        = make([]Example, 0)
 	parseExamples   sync.Once
 	archives        = singleflight.Group{}
 )
 
 type Example struct {
-	ID          string `json:"id"`
-	URL         string `json:"url"`
-	Name        string `json:"name"`
-	Description string `json:"description"`
-	Markdown    string `json:"markdown"`
+	ID            string `json:"id"`
+	URL           string `json:"url"`
+	Name          string `json:"name"`
+	Description   string `json:"description"`
+	Markdown      string `json:"markdown"`
+	DirectoryPath string `json:"directory_path"`
 }
 
-const rootDir = "templates"
+const rootDir = "quickstart"
 
-// List returns all embedded examples.
-func List() ([]Example, error) {
-	var returnError error
-	parseExamples.Do(func() {
-		files, err := fs.Sub(files, rootDir)
-		if err != nil {
-			returnError = xerrors.Errorf("get example fs: %w", err)
-		}
+// WalkForExamples will walk recursively through the examples directory and call
+// exampleDirectory() on each directory that contains a main.tf file.
+func WalkForExamples(files fs.FS, rootDir string, exampleDirectory func(path string)) error {
```

Comment:
It now will follow nested directories until it hits a `main.tf` file. So we can create w/e structure we want for these files.